pulsar/src/utils.rs

```diff
@@ -388,3 +388,37 @@ impl<'de> Deserialize<'de> for Rewards {
         Ok(Rewards(value))
     }
 }
+
+/// move data from old directory (if any) to new directory, or else just create
+/// the new directory.
+pub(crate) fn create_or_move_data(old_dir: PathBuf, new_dir: PathBuf) -> Result<()> {
+    if old_dir == new_dir {
+        bail!("This directory is already set");
+    }
+
+    if !new_dir.starts_with("/") {
+        bail!("New directory path must start with /");
+    }
+
+    // Create the new directory if it doesn't exist
+    if !new_dir.exists() {
+        fs::create_dir_all(&new_dir)?;
+    }
+
+    // Move contents if the old directory exists
+    if old_dir.exists() {
+        let entries = fs::read_dir(&old_dir)?;
+        for entry in entries {
+            let entry = entry?;
+            let file_name = entry.file_name();
+            fs::rename(entry.path(), new_dir.join(file_name))?;
+        }
+
+        // Check if the old directory is empty now and remove it
+        if fs::read_dir(&old_dir)?.next().is_none() {
```

Comment:
In which scenario, the old directory is non-empty?

Reply:
It’s possible that the data was manually deleted, perhaps by mistake. So, this is just a safety check.

Reply:
Sorry, I did not get you. Can you elaborate?

Reply:
Basically, if someone manually deletes by any chance, then the folder might be empty.
So, this is an additional check.